src/rest_vol.c

```diff
@@ -2732,14 +2477,258 @@ RV_copy_object_loc_info_callback(char *HTTP_response, void *callback_data_in, vo
+{
+    yajl_val           parse_tree = NULL, key_obj = NULL, link_obj = NULL;
+    const char        *parsed_link_name   = NULL;
+    char              *parsed_link_buffer = NULL;
+    H5VL_loc_by_idx_t *idx_params         = (H5VL_loc_by_idx_t *)callback_data_in;
+    hsize_t            index              = idx_params->n;
+    char             **link_name          = (char **)callback_data_out;
+    const char        *curr_key           = NULL;
+    herr_t             ret_value          = SUCCEED;
+
+    if (!link_name)
+        FUNC_GOTO_ERROR(H5E_ARGS, H5E_BADVALUE, FAIL, "given link_name ptr was NULL");
+
+    if (!HTTP_response)
+        FUNC_GOTO_ERROR(H5E_ARGS, H5E_BADVALUE, FAIL, "HTTP response buffer was NULL");
+
+    if (NULL == (parse_tree = yajl_tree_parse(HTTP_response, NULL, 0)))
+        FUNC_GOTO_ERROR(H5E_OBJECT, H5E_PARSEERROR, FAIL, "parsing JSON failed");
+
+    if (NULL == (key_obj = yajl_tree_get(parse_tree, links_keys, yajl_t_array)))
+        FUNC_GOTO_ERROR(H5E_OBJECT, H5E_PARSEERROR, FAIL, "failed to parse links");
+
+    if (key_obj->u.array.len == 0)
+        FUNC_GOTO_ERROR(H5E_OBJECT, H5E_PARSEERROR, FAIL, "parsed link array was empty");
+
+    if (index >= key_obj->u.array.len)
+        FUNC_GOTO_ERROR(H5E_OBJECT, H5E_PARSEERROR, FAIL, "requested link index was out of bounds");
+
+    switch (idx_params->order) {
+        case (H5_ITER_DEC):
+            if (NULL == (link_obj = key_obj->u.array.values[key_obj->u.object.len - 1 - index]))
+                FUNC_GOTO_ERROR(H5E_OBJECT, H5E_PARSEERROR, FAIL, "selected link was NULL");
+            break;
+
+        case (H5_ITER_NATIVE):
+        case (H5_ITER_INC):
+            if (NULL == (link_obj = key_obj->u.array.values[index]))
+                FUNC_GOTO_ERROR(H5E_OBJECT, H5E_PARSEERROR, FAIL, "selected link was NULL");
+            break;
+        case (H5_ITER_N):
+        case (H5_ITER_UNKNOWN):
+        default: {
+            FUNC_GOTO_ERROR(H5E_OBJECT, H5E_BADVALUE, FAIL, "invalid iteration order");
+            break;
+        }
+    }
+
+    /* Iterate through key/value pairs in link response to find name */
+    for (size_t i = 0; i < link_obj->u.object.len; i++) {
+        curr_key = link_obj->u.object.keys[i];
+
+        if (!strcmp(curr_key, "title"))
+            if (NULL == (parsed_link_name = YAJL_GET_STRING(link_obj->u.object.values[i])))
+                FUNC_GOTO_ERROR(H5E_OBJECT, H5E_PARSEERROR, FAIL, "failed to get link name");
+    }
+
+    if (NULL == parsed_link_name)
+        FUNC_GOTO_ERROR(H5E_OBJECT, H5E_PARSEERROR, FAIL, "server response didn't contain link name");
+
+    if (NULL == (parsed_link_buffer = RV_malloc(strlen(parsed_link_name) + 1)))
+        FUNC_GOTO_ERROR(H5E_OBJECT, H5E_PARSEERROR, FAIL, "failed to allocate memory for link name");
+
+    memcpy(parsed_link_buffer, parsed_link_name, strlen(parsed_link_name) + 1);
+
+    *link_name = parsed_link_buffer;
```

Comment:
Same comment as in other functions where a local buffer should be used and assigned to `*link_name` and the local buffer pointer should be freed on failure, not the pointer passed in.